src/python/zquantum/core/cost_function.py

```diff
@@ -314,3 +318,114 @@ def _preprocess(parameters: np.ndarray) -> np.ndarray:
         return parameters + noise
 
     return _preprocess
+
+
+def create_cost_function(
+    backend: QuantumBackend,
+    estimation_tasks_factory: EstimationTasksFactory,
+    estimation_method: EstimateExpectationValues = _by_averaging,
+    parameter_preprocessors: Iterable[ParameterPreprocessor] = None,
+) -> CostFunction:
+    """This function can be used to generate callable cost functions for parametric
+    circuits. This function is the main entry to use other functions in this module.
+
+    Args:
+        backend: quantum backend used for evaluation.
+        estimation_tasks_factory: function that produces estimation tasks from
+            parameters. See example use case below for clarification.
+        estimation_method: the estimator used to compute expectation value of target
+            operator.
+        parameter_preprocessors: a list of callable functions that are applied to
+            parameters prior to estimation task evaluation. These functions have to
+            adhere to the ParameterPreprocessor protocol.
+
+    Returns:
+        A callable CostFunction object.
+
+    Example use case:
+        target_operator = ...
+        ansatz = ...
+
+        estimation_factory = substitution_based_estimation_tasks_factory(
+            target_operator, ansatz
+        )
+        noise_preprocessor = add_noise(1e-5, seed=1234)
```

Comment:
I think we finally renamed it to `add_normal-noise`?

Reply:
Good catch